### PR TITLE
Automatically allocate huge pages in docker start script

### PR DESCRIPTION
**Pull request by pudelkoM**
I feel this should be handled in the `start-stratum.sh` script instead, so it also applies the the Debian package.

Is it possible to mount huge pages from inside the container?

### CODE REVIEW

File: stratum/hal/bin/barefoot/docker/start-stratum-container.sh [hunk 1]
```diff
@@ -20,10 +20,18 @@ if [ -d "/etc/onl" ]; then
               -v /etc/onl:/etc/onl"
 fi
 
+# Prepare the host.
+free_hugepages=$(cat /sys/kernel/mm/hugepages/hugepages-2048kB/free_hugepages)
+if [[ $free_hugepages -lt 128 ]]; then
```
Comment (bocon13) [inline stratum/hal/bin/barefoot/docker/start-stratum-container.sh:25]: Print a warning instead and exit.

**Comment by bocon13** [comment]
@pudelkoM I just pushed a change. Please take a look.

I did some local testing, but we should verify on the PDP switch as well.

**Comment by bocon13** [comment]
@bocon13 to test on hardware

**Review by pudelkoM — COMMENTED** [review]
Just tried this on a switch (Debian version). Worked both with 128 free pages and no prior setup.

**Comment by bocon13** [comment]
Found two issues: exit wasn't working on failure, sysctl was not working on the switch in Docker

Retested deb and container on hardware, and it works (when huge pages are and aren't allocated)

**Review by bocon13 — APPROVED** [review]

**Pull request merged**